controller/model/authenticator_mod_cert.go

```diff
@@ -81,13 +89,55 @@ func (module *AuthModuleCert) Process(context AuthContext) (string, error) {
 				}
 			}
 
-			return authenticator.IdentityId, nil
+			opts := x509.VerifyOptions{
+				Roots:         module.getRootPool(),
+				Intermediates: x509.NewCertPool(),
+				KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+			}
+
+			if _, err := cert.Verify(opts); err == nil {
+				return authenticator.IdentityId, nil
+			} else {
+				println(err)
+			}
 		}
 	}
 
 	return "", apierror.NewInvalidAuth()
 }
 
+func (module *AuthModuleCert) getRootPool() *x509.CertPool {
+	roots := x509.NewCertPool()
+
+	for _, caCert := range module.staticCaCerts {
+		roots.AddCert(caCert)
+	}
+
+	cas, err := module.env.GetHandlers().Ca.Query("isAuthEnabled = true and isVerified = true limit 500")
+
+	if err != nil {
+		return nil
```

Comment:
shouldn't we log this err "just in case"?

Reply:
changed when moving to a streaming call